lib/fluent/plugin/generate_hash_id_support.rb

```diff
@@ -0,0 +1,36 @@
+require 'securerandom'
+
+module Fluent
+  module Plugin
+    module GenerateHashIdSupport
+      def self.included(klass)
+        klass.instance_eval {
+          config_section :hash, param_name: :hash_config, required: false, multi: false do
+            config_param :hash_type, :enum, list: [:md5, :sha1, :sha256, :sha512], default: :sha256
+            config_param :hash_id_key, :string, default: '_id'
+          end
+        }
+      end
+
+      def hash_hex(str)
+        case @hash_config.hash_type
+        when :md5
+          Digest::MD5.hexdigest(str)
+        when :sha1
+          Digest::SHA1.hexdigest(str)
+        when :sha256
+          Digest::SHA256.hexdigest(str)
+        when :sha512
+          Digest::SHA512.hexdigest(str)
+        end
+      end
+
+      def generate_hash_id_key(record)
+        s = ""
+        s += hash_hex(SecureRandom.hex(20))
```

Comment:
Does this method adhere to the recommendations of good lucene keys as described at [1]?

[1] https://www.elastic.co/guide/en/elasticsearch/guide/current/index-doc.html

Reply:
Elasticsearch document says that:

> Autogenerated IDs are 20 character long, URL-safe, Base64-encoded GUID strings. 

I think that we can implement similar functionality as below:

```ruby
s += Base64.encode64(SecureRandom.uuid)
```

But this line generates 48 characters, though.

Reply:
I've used this implementation for now.